esmvalcore/preprocessor/_area.py

```diff
@@ -63,16 +63,29 @@ def extract_region(cube, start_longitude, end_longitude, start_latitude,
             ignore_bounds=True,
         )
         region_subset = region_subset.intersection(longitude=(0., 360.))
-        return region_subset
-    # Irregular grids
-    lats = cube.coord('latitude').points
-    lons = cube.coord('longitude').points
+    else:
+        region_subset = _extract_irregular_region(
+            cube,
+            start_longitude,
+            end_longitude,
+            start_latitude,
+            end_latitude,
+        )
+    return region_subset
+
+
+def _extract_irregular_region(cube, start_longitude, end_longitude,
+                              start_latitude, end_latitude):
+    """Extract a region from a cube on an irregular grid."""
     # Convert longitudes to valid range
     if start_longitude != 360.:
         start_longitude %= 360.
     if end_longitude != 360.:
         end_longitude %= 360.
 
+    # Select coordinates inside the region
+    lats = cube.coord('latitude').points
+    lons = cube.coord('longitude').points
```

Comment:
It's probably easy to account for negative longitudes here, right? That would help with #1141. What do you think about
```suggestion
    lons = (cube.coord('longitude').points + 360.) % 360. 
```

Reply:
To keep preprocessor functions simple, we assume that the [data follows the CMOR table conventions in all preprocessor functions](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#design-considerations), i.e. longitudes are inside the range 0 to 360 degrees. A fix like this is done already as part of the automatic fixes here: https://github.com/ESMValGroup/ESMValCore/blob/933b37b225a89aeed84029d8c83105407ed7ae5f/esmvalcore/cmor/check.py#L656

Reply:
This is not an actual requirement of the CMOR tables for irregular grids. The fix is wrong and will be removed.

Reply:
Besides, we need to take other sources like CORDEX into account.

Reply:
Hi @zklaus, this PR does not change the current behaviour of the code in master: https://github.com/ESMValGroup/ESMValCore/blob/933b37b225a89aeed84029d8c83105407ed7ae5f/esmvalcore/preprocessor/_area.py#L69
If you think this is a bug, could you make a separate issue to discuss an appropriate fix so we do not delay this PR any further?

Reply:
@zklaus Thanks for explaining, I changed it as suggested. If anything else needs changing to get this in: I'll be on holiday until next week Friday, so please feel free to make the changes yourself.